### DISCUSSION TITLE
Difference with x-data from 2 to 3?

### DISCUSSION THREAD

**Discussion by marcusjm** [alpinejs/alpine#3429] [1. Help]
I am wondering if and what changed in the handling of x-data from 2 to 3? I am using Alpine with Blazor and previously I manipulated the x-data with a variable from C#. Now however, nothing happens when I change the variable. (in 3.x).

What could possibly have changed that made this not work anymore?

`<div x-data="open:@DynamicVariable">(content)</div>`

Thanks in advance,
Marcus

**Comment by ekwoka** [comment]
what does that do?

Like how would it work?

**Reply by marcusjm** [reply]
Previously it triggered the same thing as a change in that variable via JS. I don't know why it works in 2 but not 3.

**Reply by ekwoka** [reply]
well, in 3, `x-data` is not reevaluated. So changing it's html string won't do anything. which I guess it what that does?

idk, `open:false` wouldn't be a valid expression in 2 or 3...

**Reply by marcusjm** [reply]
Yes unfortunatelly. That removes Alpine as a viable companion library for Blazor apps. Unless you continue using 2.x perhaps. Good to know the cause at least. The previous example worked fine in 2, that way you could mainpulate open/close with transitions but now I guess you can achieve the same anyway with TW improved handling.

**Reply by ekwoka** [reply]
I'd say Blazor made itself not viable before web dev before Alpine even came along.

But I'd imagine there is another way to make what you're trying to do work. Unfortunately, I'm an export in Alpine, and not Blazor, so I don't know a way to get them to work together. Other expressions like `x-init` should reevaluate every time, and you can of course create a reactive object blazor can manipulate and inject that into an alpine component.

Lots of things.

**Reply by marcusjm** [reply]
> I'd say Blazor made itself not viable before web dev before Alpine even came along.
> 
> But I'd imagine there is another way to make what you're trying to do work. Unfortunately, I'm an export in Alpine, and not Blazor, so I don't know a way to get them to work together. Other expressions like `x-init` should reevaluate every time, and you can of course create a reactive object blazor can manipulate and inject that into an alpine component.
> 
> Lots of things.

Blazor is mainly for making applications not web sites for us. All the current ones work fine (no need to go from alpine 2 to 3).

Yes but I mainly used Alpine for the transitions before. With Tw3 I have more control over transitions. What made Alpine 2.x good was that it allowed for some manipulation without using Interop and that was a big thing but now that all the other libraries have improved the need is less. Now I know the basis of that change.
